src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.netcore.cs

```diff
@@ -13064,42 +13060,75 @@ public static Vector512<T> Invoke(Vector512<T> x)
             //
             // coshf = v/2 * exp(x - log(v)) where v = 0x1.0000e8p-1
 
-            private const float LOGV = 0.693161f;
-            private const float HALFV = 1.0000138f;
-            private const float INVV2 = 0.24999309f;
+            private const double LOGV = 0.6931471805599453;
+            private const double HALFV = 1.0;
+            private const double INVV2 = 0.25;
```

Comment:
Sorry, I didn't mean to use the same constant for both `double` and `float`. Rather we should use the original values for `float` and the new separate constants for `double`.

The values for `float` have already been adjusted to account for rounding error and they will be different from the values used for `double` by design.